src/graphql/types/object/transaction-with-metadata.ts

```diff
@@ -0,0 +1,83 @@
+import dedent from "dedent"
+
+import { GT } from "@graphql/index"
+
+import { SAT_PRICE_PRECISION_OFFSET } from "@config"
+
+import Memo from "../scalar/memo"
+
+import InitiationVia from "../abstract/initiation-via"
+import SettlementViaWithMetadata from "../abstract/settlement-via-with-metadata"
+import Timestamp from "../scalar/timestamp"
+import TxDirection, { txDirectionValues } from "../scalar/tx-direction"
+import TxStatus from "../scalar/tx-status"
+import SignedAmount from "../scalar/signed-amount"
+import WalletCurrency from "../scalar/wallet-currency"
+
+import Price from "./price"
+
+const TransactionWithMetadata = GT.Object<WalletTransactionWithMetadata>({
```

Comment:
With this PR we seem to add quite a bit of complexity for the API consumer. 

why not do something like:

```
    fragment TransactionList on TransactionConnection {
    pageInfo {
      hasNextPage
      hasPreviousPage
      startCursor
      endCursor
    }
    edges {
      cursor
      node {
        __typename
        id
        status
        direction
        memo
        createdAt
        settlementAmount
        settlementFee
        settlementCurrency
        settlementPrice {
          base
          offset
          currencyUnit
          formattedAmount
        }
        initiationVia {
          __typename
          ... on InitiationViaIntraLedger {
            counterPartyWalletId
            counterPartyUsername
          }
          ... on InitiationViaLn {
            paymentHash
          }
          ... on InitiationViaOnChain {
            address
          }
        }
        settlementVia {
          __typename
          ... on SettlementViaIntraLedger {
            counterPartyWalletId
            counterPartyUsername
          }
          ... on SettlementViaLn {
            paymentSecret
          }
          ... on SettlementViaOnChain {
            transactionHash
          }
        }
      }
    }
  }
`

this for the main query, where we list all the transactions.

and for the individual query, something like:

```
    Transaction($transactionHash) {
      cursor
      node {
        __typename
        id
        status
        direction
        memo

        __preImage__
        __otherOptionalMetadata__

        createdAt
        settlementAmount
        settlementFee
        settlementCurrency
        settlementPrice {
          base
          offset
          currencyUnit
          formattedAmount
        }
        initiationVia {
          __typename
          ... on InitiationViaIntraLedger {
            counterPartyWalletId
            counterPartyUsername
          }
          ... on InitiationViaLn {
            paymentHash
          }
          ... on InitiationViaOnChain {
            address
          }
        }
        settlementVia {
          __typename
          ... on SettlementViaIntraLedger {
            counterPartyWalletId
            counterPartyUsername
          }
          ... on SettlementViaLn {
            paymentSecret
          }
          ... on SettlementViaOnChain {
            transactionHash
          }
        }
      }
    }
  }
`

the idea behind it is that we would not introduce a new type `WalletTransactionWithMetadata` but rather use the inherent property of Graphql to fetch, or not fetch, the individual metadata as needed.

for this to work, some of the logic for the additional/optional metadata would have to be in the graphql resolver. I'm not sure how this may interact with our domain logic yet (example here: https://stackoverflow.com/questions/61060046/graphql-only-fetch-if-certain-field-is-requested-in-query)